roles/init_pd_topo/templates/init_pd_topo.sh.j2

```diff
@@ -0,0 +1,47 @@
+#!/bin/bash
+set -e
+
+{% set all_pd = [] -%}
+{% set pd_hosts = groups.pd_servers %}
+{% for host in pd_hosts -%}
+  {% set pd_ip = hostvars[host].ansible_host | default(hostvars[host].inventory_hostname) -%}
+  {% set pd_port = hostvars[host].pd_client_port -%}
+  {% set pd_path = hostvars[host].deploy_dir -%}
+  {% set _ = all_pd.append("%s:%s%s" % (pd_ip, pd_port, pd_path)) -%}
+{% endfor -%}
+{% set all_tidb = [] -%}
+{% set tidb_hosts = groups.tidb_servers %}
+{% for host in tidb_hosts -%}
+  {% set tidb_ip = hostvars[host].ansible_host | default(hostvars[host].inventory_hostname) -%}
+  {% set tidb_port = hostvars[host].tidb_port -%}
+  {% set tidb_path = hostvars[host].deploy_dir -%}
+  {% set _ = all_tidb.append("%s:%s%s" % (tidb_ip, tidb_port, tidb_path)) -%}
```

Comment:
how would you parse the string when you put these together?

Reply:
Parameter values are strings separated by `,`.